src/util/kaldi-table-inl.h

```diff
@@ -248,49 +271,66 @@ template<class Holder>  class SequentialTableReaderScriptImpl:
   }
 
   virtual ~SequentialTableReaderScriptImpl() {
-    if (state_ == kError)
-      KALDI_ERR << "Reading script file failed: from scp "
-                << PrintableRxfilename(script_rxfilename_);
-    // If you don't want this exception to be thrown you can
-    // call Close() and check the status.
-    if (state_ == kLoadSucceeded)
-      holder_.Clear();
+    if (this->IsOpen() && !Close())
+      KALDI_ERR << "TableReader: reading script file failed: from scp "
+                      << PrintableRxfilename(script_rxfilename_);
   }
  private:
   bool LoadCurrent() {
```

Comment:
document the possible states at the end of this function, and the associated return statuses.